### PR TITLE
fix(suite-desktop): use native titlebar actions

### LINKED ISSUE #4243: Desktop app titlebar actions do not work when app has no focus
**Describe the bug**
If the desktop app does not have focus, it is not possible to close it (minimise it or use fullscreen). However, other macOS apps can be closed without focus. Moreover, if I click on actions, it does not give focus to the window.

https://user-images.githubusercontent.com/33235762/132324912-44b6b648-8e1f-43f7-a5b0-4b7b9b74dc30.mov

**Steps to reproduce:**
1. Open desktop app on macOS
2. Give focus to another app
3. Try to close Suite

**Expected result:**
I am able to close unfocused app.

**Note(s):**
I think that we do something wrong with the whole macOS desktop app window, because I am unable to use Magnet app (https://apps.apple.com/cz/app/magnet/id441258766) to move and resize window using key shortcuts.

### PR DESCRIPTION
**Pull request by tomasklim**
Closes #4243

After consultation with Ondra, we decided that we will remove custom titlebar. This means that we will not have Suite icon there anymore. Every platform will now have its native titlebar and its actions and will behave the way the users of specific platforms are used to.

In Design 2.0, we will try to use modern titlebars as e.g. spotify or slack has.
Screenshot 2021-10-28 at 10 28 20

Screenshot 2021-10-28 at 10 30 37

### CODE REVIEW

**Comment by tomasklim** [comment]
**QA manual:**

1. Failed transaction should have X (cross) icon in notification
2. Notifications on desktop and web should be on identical place
3. Modals on desktop and web should be in the centre of app  (as in previous versions)
4. Desktop app should have borders on windows and linux. E.g. you put Trezor app (light mpde) window over white browser page and you should see borders of Trezor app.
5. Reconnect Device Promt should be on the same place in both web and desktop apps (as in previous versions)
6. Titlebar and also its action icons should be working. (e.g. Dragging, maximization using double click)
7. Titlebar respects app theme. (also when app restarted)

File: packages/suite/src/components/suite/Modal/index.tsx [hunk 1]
```diff
@@ -1,13 +1,6 @@
...
-import { DESKTOP_WRAPPER_BORDER_WIDTH } from '@suite-constants/layout';
-import { isDesktop, isMacOs } from '@suite-utils/env';
 
-const Modal = (props: ModalProps) => (
-    <TrezorModal
-        {...props}
-        desktopBorder={isDesktop() && !isMacOs() ? DESKTOP_WRAPPER_BORDER_WIDTH : undefined}
-    />
-);
+const Modal = (props: ModalProps) => <TrezorModal {...props} />;
```
Comment (marekrjpolak) [inline packages/suite/src/components/suite/Modal/index.tsx]: This component became useless and could be replaced directly by Modal from `@trezor/components`.

File: packages/suite/src/components/suite/Settings/components/Theme.tsx [hunk 2]
```diff
@@ -46,9 +46,11 @@ const Theme = () => {
     const onChange = ({ value }: typeof THEME_OPTIONS[number]['options'][number]) => {
         if ((value === 'auto') !== autodetectTheme) {
             setAutodetect({ theme: !autodetectTheme });
+            window.desktopApi?.themeSystem();
```
Comment (marekrjpolak) [inline packages/suite/src/components/suite/Settings/components/Theme.tsx]: The `themeSystem()` method is called even if autodetection is currently turning off.

File: packages/suite/src/support/suite/Autodetect.tsx [hunk 3]
```diff
@@ -20,7 +20,11 @@ const Autodetect = () => {
     });
 
     useEffect(() => {
-        if (!storageLoaded || !autodetectTheme) return;
+        if (!storageLoaded) return;
+        if (storageLoaded && !autodetectTheme) {
```
Comment (marekrjpolak) [inline packages/suite/src/support/suite/Autodetect.tsx]: The `storageLoaded` is redundant on this line because of the previous line.

**Review by marekrjpolak — CHANGES_REQUESTED** [review]
Looks good, but I'm not able to change the titlebar theme on NixOS and cannot tell if it's NixOS issue or not. Testing on other OSes is needed.

**Comment by tomasklim** [comment]
@marekrjpolak fixed and squashed

File: packages/suite-desktop/src-electron/modules/theme.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        if (theme === 'light') {
+            logger.info('theme', `Manually setting title bar to light theme.`);
+
+            nativeTheme.themeSource = 'light';
+            setThemeSettings('light');
+        } else {
+            logger.info('theme', `Manually setting title bar to dark theme.`);
+
+            nativeTheme.themeSource = 'dark';
+            setThemeSettings('dark');
```
Comment (marekrjpolak) [inline packages/suite-desktop/src-electron/modules/theme.ts]: Is it necessary to repeat identical code block for `light`/`dark` mode (and also for `system` mode below)?

**Review by marekrjpolak — CHANGES_REQUESTED** [review]

**Comment by tomasklim** [comment]
Dark/Light mode problems:
**macOS**: 
- everything works fine, title bar color changes on system/app settings

**Linux** 
- everything should work fine, except title bar color changes only when app is restarted

**Windows**
- everything should work fine, title bar color is light unless user changes it here and chooses a color
Screenshot 2021-11-02 at 11 07 51

File: packages/suite-desktop/src-electron/modules/theme.ts (hunk 4 above)
Comment (tomasklim) [reply packages/suite-desktop/src-electron/modules/theme.ts]: c0593d9e817a88b8524e9bb6c1f05ac41fa3aecd

**Review by marekrjpolak — APPROVED** [review]

**Pull request merged**